src/sentry/spans/consumers/process/factory.py

```diff
@@ -30,20 +30,21 @@ def _process_relay_span_v1(relay_span: Mapping[str, Any]) -> SpanEvent:
     start_timestamp = datetime.utcfromtimestamp(relay_span["start_timestamp"])
     end_timestamp = datetime.utcfromtimestamp(relay_span["timestamp"])
     snuba_span: SpanEvent = SpanEvent(
+        duration_ms=max(
+            int((end_timestamp - start_timestamp).total_seconds() * 1e3),
+            0,
+        ),
         exclusive_time_ms=int(relay_span.get("exclusive_time", 0)),
         is_segment=relay_span.get("is_segment", False),
         organization_id=relay_span["organization_id"],
         parent_span_id=relay_span.get("parent_span_id", "0"),
         project_id=relay_span["project_id"],
+        received=relay_span["received"],
```

Comment:
Is received a required field in the schema? i would assume that it would have been added as an optional field to begin with since it does not exist currently. In that case you might want to use a get method to retrieve the field.

Reply:
It is indeed optional for now and I will set it as required a few days after we deployed the Relay change adding it to the payload.

Reply:
In that case do you want to change this line to something like below to avoid making the consumer crash?
```suggestion
        received=relay_span.get("received", current_timestamp),
```